sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * The name of a metric.
+ */
+@Experimental(Kind.METRICS)
+@AutoValue
+public abstract class MetricName {
+
+  /** The inNamespace that declared this metric. */
+  public abstract String getNamespace();
+
+  /**
+   * The name of this metric.
+   */
+  public abstract String getName();
+
+  public static MetricName named(String namespace, String name) {
+    return new AutoValue_MetricName(namespace, name);
+  }
+
+  public static MetricName named(Class<?> namespace, String name) {
+    return new AutoValue_MetricName(namespace.getSimpleName(), name);
```

Comment:
Seems reasonable -- we still need to nail down how the namespace should be setup in Python, so I don't want to get too hung up on this. It seems like the full path is reasonable though, switched to `.getName()`.